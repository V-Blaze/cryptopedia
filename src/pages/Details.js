import React from 'react';
import { useLocation } from 'react-router-dom';

const Details = () => {
  const location = useLocation();
  const {
    exchangeId, name, rank, percentTotalVolume, volumeUsd, tradingPairs, socket, exchangeUrl,
  } = location.state;

  const money = new Intl.NumberFormat('en-US',
    { style: 'currency', currency: 'USD' });

  return (
    <div>
      <div className="hero-details">
        <div className="inner-hero px-4">
          <img
            src={`https://www.google.com/s2/favicons?sz=64&domain_url=${exchangeUrl}`}
            alt="logo"
            className="w-20 h-20"
          />
          <div className="text-2xl text-white flex flex-col gap-2">
            <span className=" font-semibold">
              Name:
              <span className="ml-2 text-red-300 font-bold text-4xl">{name}</span>
            </span>
            <span className=" font-semibold">
              Rank:
              <span className="ml-2 text-red-300 font-bold text-4xl">{rank}</span>
            </span>
          </div>
        </div>
      </div>
      <div className=" flex flex-col">
        <h3 className="bg-gradient-to-r hover:bg-gradient-to-l from-sky-500 to-indigo-500 self-center text-3xl w-full text-center font-bold text-white py-2 rounded-b-lg">Exchange Details</h3>
        <ul className="flex flex-col p-5 gap-1 text-white font-semibold text-xl">
          <li className="flex justify-between bg-blue-800 py-3 rounded-md px-1">
            <span>Trading Volum percent</span>
            <span>
              {Number(percentTotalVolume).toFixed(3)}
              %
            </span>
          </li>
          <li className="flex justify-between bg-blue-900 py-3 rounded-md px-1">
            <span>Trading Volum USD:</span>
            <span>
              {money.format(volumeUsd)}
            </span>
          </li>
          <li className="flex justify-between bg-blue-800 py-3 rounded-md px-1">
            <span>No. of Trading Pairs</span>
            <span>{tradingPairs}</span>
          </li>
          <li className="flex justify-between bg-blue-900 py-3 rounded-md px-1">
            <span>Socket</span>
            <span>{socket}</span>
          </li>
          <li className="flex justify-between bg-blue-800 py-3 rounded-md px-1">
            <span>Website</span>
            <span>{exchangeUrl}</span>
          </li>
          <li className="flex justify-between bg-blue-900 py-3 rounded-md px-1">
            <span>Exchange ID</span>
            <span>{exchangeId}</span>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Details;
