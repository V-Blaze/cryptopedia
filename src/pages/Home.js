import React, { useState, useEffect } from 'react';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import ExchangeLists from '../components/ExchangeLists';
import { filterCountry } from '../utils';

const Home = () => {
  const navigate = useNavigate();
  const exchanges = useSelector((state) => state.exchanges);
  const [cryptoExchanges, setCryptoExchanges] = useState([]);

  useEffect(() => {
    setCryptoExchanges(exchanges);
  }, [exchanges]);

  const handleSelection = (country) => {
    const filtered = filterCountry(country, exchanges);
    setCryptoExchanges(filtered);
  };

  const handleSearch = (text) => {
    const filteredExchanges = exchanges.filter((ex) => (
      ex.name.toLowerCase().includes(text.toLowerCase())));
    setCryptoExchanges(filteredExchanges);
  };

  const showDetails = (id) => {
    const item = exchanges.filter((exchange) => exchange.exchangeId === id);
    navigate('/details', { state: item[0] });
    console.clear();
  };

  return (
    <div>
      <div className="bg-[#3775e8] h-[15rem] flex flex-col gap-3 justify-center items-center">
        <div>
          <span className=" text-xl font-semibold italic text-gray-300 ">
            Total Exchanges:
            {`  ${cryptoExchanges.length}`}
          </span>
        </div>
        <div className="w-full flex flex-col items-center gap-2">
          <input
            type="text"
            placeholder="Search Exchange"
            onChange={(e) => handleSearch(e.target.value)}
            className="w-[80%] rounded-lg px-2 py-2 bg-blue-900 text-white"
          />
          <form className="w-[60%]">
            {/* <label htmlFor="countries">Choose an countries:</label> */}
            <select
              id="countries"
              name="countries"
              onChange={(e) => handleSelection(e.target.value)}
              className="w-full rounded-md p-1 bg-blue-900 text-white"
            >
              <option value="Select A Country">Select A Country</option>
              <option value="allCountries">All Countries</option>
              <option value="Australia">Australia</option>
              <option value="China">China</option>
              <option value="EU">EU</option>
              <option value="Indonesia">Indonesia</option>
              <option value="Japan">Japan</option>
              <option value="Russia">Russia</option>
              <option value="Seychelles">Seychelles</option>
              <option value="Singapore">Singapore</option>
              <option value="South_Korea">South Korea</option>
              <option value="UK">UK</option>
              <option value="US">US</option>
              <option value="otherCountries">Other Countries</option>
            </select>
          </form>
        </div>
      </div>
      <div className="exchange-container mb-9 ">
        {cryptoExchanges.map((exchange) => (
          <ExchangeLists
            key={exchange.exchangeId}
            exchange={exchange}
            showDetails={showDetails}
          />
        ))}
      </div>
      <div className=" fixed bottom-0 mt-4 w-full bg-[#4369B2] h-9 flex justify-center items-center text-white text-xl">
        <span>CryptoPedia Copywrite V-Blaze</span>
      </div>
    </div>
  );
};

export default Home;
