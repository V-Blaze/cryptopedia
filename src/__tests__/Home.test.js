import { screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import { Home } from '../pages';
import renderWithProviders from '../redux/Exchanges/exchange-test-utils';

it('snapshots', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('testing the heading inside home component', async () => {
  renderWithProviders(
    <Router>
      <Home />
    </Router>,
  );

  // should show no user initially, and not be fetching a user
  expect(screen.getByText(/CryptoPedia/i)).toBeInTheDocument();
});
