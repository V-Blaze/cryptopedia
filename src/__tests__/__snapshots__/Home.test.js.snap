// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div>
  <div
    className="bg-[#3775e8] h-[15rem] flex flex-col gap-3 justify-center items-center"
  >
    <div>
      <span
        className=" text-xl font-semibold italic text-gray-300 "
      >
        Total Exchanges:
          0
      </span>
    </div>
    <div
      className="w-full flex flex-col items-center gap-2"
    >
      <input
        className="w-[80%] rounded-lg px-2 py-2 bg-blue-900 text-white"
        onChange={[Function]}
        placeholder="Search Exchange"
        type="text"
      />
      <form
        className="w-[60%]"
      >
        <select
          className="w-full rounded-md p-1 bg-blue-900 text-white"
          id="countries"
          name="countries"
          onChange={[Function]}
        >
          <option
            value="Select A Country"
          >
            Select A Country
          </option>
          <option
            value="allCountries"
          >
            All Countries
          </option>
          <option
            value="Australia"
          >
            Australia
          </option>
          <option
            value="China"
          >
            China
          </option>
          <option
            value="EU"
          >
            EU
          </option>
          <option
            value="Indonesia"
          >
            Indonesia
          </option>
          <option
            value="Japan"
          >
            Japan
          </option>
          <option
            value="Russia"
          >
            Russia
          </option>
          <option
            value="Seychelles"
          >
            Seychelles
          </option>
          <option
            value="Singapore"
          >
            Singapore
          </option>
          <option
            value="South_Korea"
          >
            South Korea
          </option>
          <option
            value="UK"
          >
            UK
          </option>
          <option
            value="US"
          >
            US
          </option>
          <option
            value="otherCountries"
          >
            Other Countries
          </option>
        </select>
      </form>
    </div>
  </div>
  <div
    className="exchange-container mb-9 "
  />
  <div
    className=" fixed bottom-0 mt-4 w-full bg-[#4369B2] h-9 flex justify-center items-center text-white text-xl"
  >
    <span>
      CryptoPedia Copywrite V-Blaze
    </span>
  </div>
</div>
`;
